core/include/scipp/core/value_and_variance.h

```diff
@@ -78,6 +78,24 @@ template <class T> constexpr auto abs(const ValueAndVariance<T> a) noexcept {
   return ValueAndVariance{abs(a.value), a.variance};
 }
 
+template <typename T> constexpr auto exp(const ValueAndVariance<T> a) noexcept {
+  using std::exp;
+  const auto val = exp(a.value);
+  return ValueAndVariance(val, val * val * a.variance);
+}
+
+template <typename T> constexpr auto log(const ValueAndVariance<T> a) noexcept {
+  using std::log;
+  return ValueAndVariance(log(a.value), a.variance / (a.value * a.value));
+}
+
+template <typename T> constexpr auto log10(const ValueAndVariance<T> a) noexcept {
+  using std::log10;
+  constexpr auto log_of_10 = static_cast<T>(2.3025850929940456840363388613113);
+  const auto x = a.value * log_of_10;
+  return ValueAndVariance(log10(a.value), a.variance / (x * x));
+}
+
```

Comment:
Presumably you've had to add these here alongside your other changes because you otherwise get compilation errors from transform? While these additions in principle seem ok to me, you could implement later - you can also disable support for ValuesAndVariance at the transform level via transform flags.

Reply:
This is a limitation similar to the one I discuss below regarding specifying output types: We cannot use `assign_op` with position-dependent flags such as `expect_no_variance_arg<N>`, since `N` changes to `N+1` when `assign_op` is used to wrap the operator.